src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs

```diff
@@ -2472,6 +2472,186 @@ static void Main()
         }
 
         [Fact]
+        public void RefReadonlyConditional()
+        {
+            var comp = CompileAndVerify(@"
+using System;
+
+struct S
+{
+    public int X;
+    public S(int x) => X = x;
+
+    public void Mutate() => X++;
+}
+
+class C
+{
+    static void Main()
+    {
+        S local1 = new S(0);
+        S local2 = new S(0);
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        bool condition = false;
+
+        (condition ? ref local1 : ref local2).Mutate();
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        (condition ? ref local1 : ref local2).X = 0;
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        ref readonly S ro1 = ref local1;
+        ref readonly S ro2 = ref local2;
+
+        (condition ? ref local1 : ref ro2).Mutate();
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        (condition ? ref ro1 : ref local2).Mutate();
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        (!condition ? ref local1 : ref ro2).Mutate();
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
+
+        (!condition ? ref ro1 : ref local2).Mutate();
+        Console.WriteLine(local1.X);
+        Console.WriteLine(local2.X);
```

Comment:
Consider adding `Console.WriteLine(ro1.X); Console.WriteLine(ro2.x);`, perhaps once at the end of this method.